scripts/ci/hasFileChanged.mjs

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env node
+import {setOutput} from '@actions/core';
+import {context} from '@actions/github';
+import {minimatch} from 'minimatch';
+import {execSync} from 'node:child_process';
+
+function getBaseHeadSHAs() {
+  switch (context.eventName) {
+    case 'pull_request':
+      return {
+        base: context.payload.pull_request.base.sha,
+        head: context.payload.pull_request.head.sha,
+      };
+    case 'merge_group':
+      return {
+        base: context.payload.merge_group.base_sha,
+        head: context.payload.merge_group.head_sha,
+      };
+  }
+}
+
+function getChangedFiles(from, to) {
+  return execSync(`git diff --name-only ${from}..${to}`, {
+    stdio: 'pipe',
+    encoding: 'utf-8',
+  });
+}
+
+function getPatterns() {
+  return process.argv.slice(3);
+}
+
+function checkPatterns(files, patterns) {
```

Comment:
Not a big deal at all, but you don't even need to check all files against all patterns; as soon as there's a single match you can return true.

Reply:
This is what the code does already tho 😄

Reply:
Indeed, I'm silly 🙃